queue_job/models/queue_job.py

```diff
@@ -398,13 +398,19 @@ def autovacuum(self):
         """
         for channel in self.env["queue.job.channel"].search([]):
             deadline = datetime.now() - timedelta(days=int(channel.removal_interval))
+            subchannels = self.env["queue.job.channel"].search(
+                [("parent_id", "=", channel.id)]
+            )
             while True:
                 jobs = self.search(
                     [
                         "|",
                         ("date_done", "<=", deadline),
                         ("date_cancelled", "<=", deadline),
+                        "|",
                         ("channel", "=", channel.complete_name),
+                        ("channel", "ilike", channel.complete_name + ".%"),
```

Comment:
```suggestion
                        ("channel", "=like", channel.complete_name + ".%"),
```
Otherwise it will add % before and after.